### PR TITLE
MGMT-1247 Assisted-service diff tool

### PR DESCRIPTION
**Pull request by oshercc**

### CODE REVIEW

File: tools/version_docomentaion.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import argparse
+import logging
+import shutil
+import os
+import re
+
+logging.basicConfig(format='%(asctime)s %(message)s')
+logging.getLogger().setLevel(logging.INFO)
+
+REPO_URL = "https://github.com/openshift/assisted-service.git"
```
Comment (filanov) [inline tools/version_docomentaion.py:11]: I think that we need it for all the repos, so maybe it should be a parameter?

File: tools/version_docomentaion.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+
+def process_logs(git_logs_line, version_documentation_list):
+    for line in git_logs_line:
+        line = line.strip().decode("utf-8")
+        if is_line_metadata(line):
+            continue
+        line = line.strip('* ')
+
+        if re.match(valid_commit_regex, line):
```
Comment (gamli75) [inline tools/version_docomentaion.py:50]: why are you checking if it's a valid commit regex? we already saw that we have problems when we cherry-pick commits in a branch - please verify that it's working on a branch

File: tools/version_docomentaion.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        f.write(version_documentation)
+
+
+def is_line_metadata(line):
+    line_starts_with = ['commit', 'Author', 'Date:']
+    for meta_prefix in line_starts_with:
+        if line.startswith(meta_prefix):
+            return True
+    if line == '':
+        return True
```
Comment (gamli75) [inline tools/version_docomentaion.py:80]: I think it's should be false

File: tools/version_docomentaion.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+args = parser.parse_args()
+
+
+def main():
+    version_documentation_list = list()
+
+    if not os.path.exists(args.documentation_dir):
+        os.makedirs(args.documentation_dir)
+
+    os.mkdir("temp")
```
Comment (gamli75) [inline tools/version_docomentaion.py:32]: why do you need this temp dir?

File: tools/version_docomentaion.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        line = line.strip('* ')
+
+        if re.match(valid_commit_regex, line):
+            version_documentation_list.append(line)
+
+
+def get_versions_log():
+    subprocess.check_output("git clone {}".format(REPO_URL), shell=True, cwd="temp")
+    raw_log = subprocess.check_output("git log {tagS}...{tagE} ".format(tagS=args.from_version, tagE=args.to_version),
+                                      shell=True, cwd="temp/assisted-service")
```
Comment (gamli75) [inline tools/version_docomentaion.py:57]: same remark as @filanov - use repo parameter

File: tools/version_docomentaion.py (hunk 3 above)
Comment (oshercc) [reply tools/version_docomentaion.py:83]: nope

File: tools/version_docomentaion.py (hunk 4 above)
Comment (oshercc) [reply tools/version_docomentaion.py:34]: we clone the repo to it to get the logs for processing

File: tools/version_docomentaion.py (hunk 2 above)
Comment (oshercc) [reply tools/version_docomentaion.py:52]: that's exactly why I take only the validated lines, the metadata is not documented

**Pull request merged**